packages/react-core/src/components/NumberInput/NumberInput.tsx

```diff
@@ -59,6 +59,18 @@ const defaultKeyDownHandler = (args: DefaultKeyDownHandlerArgs) => (event: React
   }
 };
 
+const normalizeBetween = ({ value, min, max }: Pick<NumberInputProps, 'value' | 'min' | 'max'>) => {
+  let result = value;
+  if (min !== undefined && max !== undefined) {
+    result = Math.max(Math.min(value, max), min);
+  } else if (min === undefined && value <= min) {
+    result = min;
+  } else if (max === undefined && value >= max) {
+    result = max;
+  }
+  return result;
+};
+
```

Comment:
Are the two `else if` conditions correct here? I wasn't sure if they should be `max === undefined && value <= min` and vice versa or not.

Reply:
It is a little confusing, if `min` is undefined then `value <= min` wouldn't make a ton of sense.

Reply:
maybe I can tweak this a little bit but the idea is simple. there are 4 different cases:
1) min and max aren't defined: in that case return we don't do anything but return the value.
2) min and max are defined: check that the value is in between `min` and `max` and if not return `max` or `min`.
3) only min `defined`: if value is smaller than `min` return `min`
4) only max `defined`: if value is bigger than `max` return `max`

Will `switch-case` make it better?

Reply:
I think the `if...else` blocks are fine. Like @kmcfaul mentioned, though, the conditions don't seem to make sense when compared to the 4 cases you've laid out. For example the second condition is checking whether min is undefined, but also whether the value is less than/equal to min (which should be undefined).

When testing out on the build page, if I pass in a value to the `max` prop, but don't pass anything into the `min` prop, the value doesn't clamp to that max value like it's supposed to.